Makefile

```diff
@@ -14,143 +14,43 @@
-	$(docker-compose) up -d pgadmin
-	$(docker-compose) up -d indexer
-	$(docker-compose) up -d matcher
-	$(docker-compose) up -d notifier
-	$(docker-compose) up -d swagger-ui
-
-.PHONY: local-dev-up-with-quay
-local-dev-up-with-quay: vendor
-	## clair ##
-	$(docker-compose) up -d traefik
-	$(docker-compose) up -d jaeger
-	$(docker-compose) up -d prometheus
-	$(docker-compose) up -d grafana
-	$(docker-compose) up -d rabbitmq
-	$(docker-compose) up -d activemq
-	$(docker-compose) up -d clair-db
-	$(docker) exec -it clair-db bash -c 'while ! pg_isready; do echo "waiting for clair postgres"; sleep 2; done'
-	$(docker-compose) up -d pgadmin
-	$(docker-compose) up -d indexer-quay
-	$(docker-compose) up -d matcher
-	$(docker-compose) up -d notifier
-	$(docker-compose) up -d swagger-ui
-	## quay ##
-	$(docker-compose) up -d redis
-	$(docker-compose) up -d quay-db
-	$(docker) exec -it quay-db bash -c 'while ! pg_isready; do echo "waiting for quay postgres"; sleep 2; done'
-	$(docker) exec -it quay-db /bin/bash -c 'echo "CREATE EXTENSION IF NOT EXISTS pg_trgm" | psql -d quay -U quay'
-	$(docker-compose) up -d quay
+local-dev/clair/quay.yaml: local-dev/clair/config.yaml
+	sed '/target:/s,webhook-target/,clair-quay:8443/secscan/notification,' <$< >$@
 
-.PHONY: local-dev-restart-quay
-local-dev-restart-quay: 
-	$(docker-compose) up -d --force-recreate quay
-
-# starts a local dev environment for testing notifier
-# the notifier will create a notification on very notifier.poll_interval value in the local dev configuration.
-# 
-# the notifier will deliver the notification to the configured deliverer in the local dev configuration. 
-# the default deliverer is rabbitmq/amqp
+# Start a local development environment.
 #
-# local dev configuration can be found in "./local-dev/clair/config.yaml"
-.PHONY: local-dev-notifier-test
-local-dev-notifier-test: vendor
-	$(docker-compose) up -d traefik
-	$(docker-compose) up -d jaeger
-	$(docker-compose) up -d prometheus
-	$(docker-compose) up -d rabbitmq
-	$(docker-compose) up -d activemq
-	$(docker-compose) up -d clair-db
-	$(docker) exec -it clair-db bash -c 'while ! pg_isready; do echo "waiting for postgres"; sleep 2; done'
-	$(docker-compose) up -d notifier-test-mode
-	$(docker-compose) up -d swagger-ui
-
-.PHONY: local-dev-notifier-test-restart
-local-dev-notifier-test-restart: vendor
-	$(docker-compose) up -d --force-recreate notifier-test-mode
-
+# Each service runs in its own container to test service-to-service
+# communication.  Local dev configuration can be found in
+# "./local-dev/clair/config.yaml"
+.PHONY: local-dev
+local-dev: vendor
+	$(docker-compose) up -d
+
+compose_profiles = $(patsubst %,local-dev-%,quay notifier debug)
+.PHONY: $(compose_profiles)
+local-dev-%: vendor
+	$(docker-compose) --profile $* up -d
+
+local-dev-quay: local-dev/clair/quay.yaml vendor
+	CLAIR_CONFIG=$(<F) $(docker-compose) --profile quay up -d
```

Comment:
you think cleaning up the `local-dev/clair/quay.yaml` file here is a bit overreaching?

Reply:
What do you mean "cleaning up?" It needs to exist while things are running, and needs to be slightly different from the standard config. It being a proper target and dependency at least makes sure it gets updated as the standard config changes.